File: sprint2/JS/lesson6code1.js
function get_node_by_index(node, index) {
  while (index) {
    node = node.next;
    index -= 1;
  }
  return node;
}

function insert_node(head, index, value) {
  const new_node = new Node(value);
  if (index === 0) {
    new_node.next = head;
    return new_node;
  }
  const previous_node = get_node_by_index(head, index - 1);
  new_node.next = previous_node.next;
  previous_node.next = new_node;
  return head;
}

let node = n1, index = 2, value = 'new_node';
let head = insert_node(node, index, value);
print_linked_list(head);

File: sprint2/By lessons/lesson6code1/lesson6code1.swift
func getNodeByIndex<T>(_ node: Node<T>?, index: Int) -> Node<T>? {
    var currentNode = node
    var currentIndex = index
    while currentIndex > 0 {
        currentNode = currentNode?.next
        currentIndex -= 1
    }
    return currentNode
}

func insertNode<T>(_ head: Node<T>?, index: Int, value: T) -> Node<T>? {
    let newNode = Node(value: value)
    if index == 0 {
        newNode.next = head
        return newNode
    }
    let previousNode = getNodeByIndex(head, index: index - 1)
    newNode.next = previousNode?.next
    previousNode?.next = newNode
    return head
}

let index = 2
let value = "new_node"
let head = insertNode(n1, index: index, value: value)
printLinkedList(head)
//first -> second -> new_node -> third -> None